flux-refineck/src/type_env/paths_tree.rs

```diff
@@ -351,7 +351,8 @@ impl PathsTree {
                     let (boxed, _) = box_args(substs);
                     ty = boxed.clone();
                 }
-                (Field(field), TyKind::Indexed(BaseTy::Tuple(tys), _)) => {
+                (Field(field), TyKind::Indexed(BaseTy::Tuple(tys), _))
+                | (Field(field), TyKind::Indexed(BaseTy::Closure(_, tys), _)) => {
```

Comment:
If we make `BaseTy::Closure` take a `Substs` we should construct a `ClosureSubsts` here and use a version of [tupled_upvars_ty](https://doc.rust-lang.org/nightly/nightly-rustc/rustc_middle/ty/struct.ClosureSubsts.html#method.tupled_upvars_ty).